src/expr/src/relation.proto

```diff
@@ -71,11 +73,17 @@ message ProtoAggregateFunc {
         bool ignore_nulls = 4;
     };
 
-    message ProtoWindowFrame {
+    message ProtoFramedWindowFunc {
         ProtoColumnOrders order_by = 1;
         mz_expr.relation.ProtoWindowFrame window_frame = 2;
     }
 
+    message ProtoWindowAggregate {
```

Comment:
I saw the PR section after making this comment, feel free to ignore this.

Reply:
This was the error msg: https://buildkite.com/materialize/tests/builds/65486#018b154d-ff92-42a9-b61e-151d86c3309a

We were also wondering about this with @teskje why exactly would this be flagged by the linting. We were not sure, but our best guess is that it is because the linting is somehow set to also include compatibility in not just protobuf's binary format, but also in text formats, [where it would indeed be backward-incompatible](https://github.com/protocolbuffers/protobuf/issues/3949). Maybe we could relax the linting a bit? (Although, we do actually use the text format for EXPLAIN AS JSON, right? But we might not want to be so strict about backwards compatibility there for now.)

Reply:
> We were not sure, but our best guess is that it is because the linting is somehow set to also include compatibility in not just protobuf's binary format, but also in text formats

My understanding was that this was covered by [the `WIRE_JSON` level, not `WIRE`](https://buf.build/docs/breaking/rules#file-and-package). @nrainer-materialize any idea why this change would cause `buf` to complain?

> But we might not want to be so strict about backwards compatibility there for now.

None of the `EXPLAIN` formats are stable and we explicitly flag this.

Reply:
I am also surprised that `WIRE` complains about this because it should allow renaming messages... The error message is about a `changed type`, is buf maybe unable to detect that this was just a rename?

Reply:
A shot in the dark, but maybe `buf` has a bug where it is confused by the same thing as I was confused when reading the original code, namely that there were two messages named `ProtoWindowFrame`. So maybe it mistakenly thought that the type changed from the top-level `ProtoWindowFrame` (which is still present in the code) to the `ProtoFramedWindowFunc` (which was earlier also named `ProtoWindowFrame`).